### ISSUE TITLE
[Pyre setup.sh] Terminal closes while building from source

### ISSUE THREAD

**Issue by ProgrammingMuffin** [facebook/pyre-check#800]
#### Description
I wanted to setup pyre on local and I started following this guide https://pyre-check.org/docs/installation/#building-from-source . When I run `./scripts/setup.sh --local`, the terminal closes by itself which is a little unusual because a process shouldn't kill the terminal itself. 

I tried running `./scripts/setup.sh --local > setupdump.txt` and after the terminal closed, I re-opened the terminal and tried to see what logs are present inside that setupdump.txt file but turns out that there's nothing there (likely because logs might not have been persisted and the terminal closed before that)

I am using Fedora release 36, Python 3.10.11, ocamlopt 4.14.0, Dune 3.7.1
Is this expected behavior, has anyone faced this or have any suggestions on how I must debug this to get the logs? Thank you for your help, appreciated.

**Comment by ProgrammingMuffin** [comment]
I did some digging, and used `script output.txt` command to write terminal stdout to a file and ran the same command. I was able to extract some logs.

`[2023-10-01 14:48:50,951] [INFO] ['opam', '--cli=2.1', 'install', '--yes', '--assume-depexts', 'base64.3.5.1', 'cmdliner.1.1.1', 'core.v0.15.1', 'core_unix.v0.15.2', 're2.v0.15.0', 'dune.3.7.1', 'yojson.2.0.2', 'jsonm.1.0.2', 'ppx_deriving_yojson.3.7.0', 'ppx_yojson_conv.v0.15.1', 'ounit2.2.2.7', 'menhir.20220210', 'lwt.5.6.1', 'lwt_ppx.2.1.0', 'ounit2-lwt.2.2.7', 'pyre-ast.0.1.9', 'mtime.1.4.0', 'errpy.0.0.9']
[NOTE] Package errpy is already installed (current version is 0.0.9).
[NOTE] Package mtime is already installed (current version is 1.4.0).
[NOTE] Package pyre-ast is already installed (current version is 0.1.9).
[NOTE] Package ounit2-lwt is already installed (current version is 2.2.7).
[NOTE] Package lwt_ppx is already installed (current version is 2.1.0).
[NOTE] Package lwt is already installed (current version is 5.6.1).
[NOTE] Package menhir is already installed (current version is 20220210).
[NOTE] Package ounit2 is already installed (current version is 2.2.7).
[NOTE] Package ppx_yojson_conv is already installed (current version is v0.15.1).
[NOTE] Package ppx_deriving_yojson is already installed (current version is 3.7.0).
[NOTE] Package jsonm is already installed (current version is 1.0.2).
[NOTE] Package yojson is already installed (current version is 2.0.2).
[NOTE] Package dune is already installed (current version is 3.7.1).
[NOTE] Package re2 is already installed (current version is v0.15.0).
[NOTE] Package core_unix is already installed (current version is v0.15.2).
[NOTE] Package core is already installed (current version is v0.15.1).
[NOTE] Package cmdliner is already installed (current version is 1.1.1).
[NOTE] Package base64 is already installed (current version is 3.5.1).
[2023-10-01 14:48:55,388] [INFO] ['make', 'dev']
abort: no repository found in '/home/spider/Documents/pyre-check/source' (.hg not found)
Scanned 0 directories
                     
Scanned 0 directories
                     
Done: 0% (0/0, 0 left) (jobs: 1)
                                
Done: 0% (0/0, 0 left) (jobs: 0)
                                
Done: 0% (0/2, 2 left) (jobs: 0)
                                
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 50% (2/4, 2 left) (jobs: 0)
                                 
Done: 50% (2/4, 2 left) (jobs: 0)
                                 
Done: 33% (2/6, 4 left) (jobs: 0)
                                 
Done: 1% (2/107, 105 left) (jobs: 0)
                                    
Done: 1% (2/107, 105 left) (jobs: 0)
                                    
Done: 26% (28/107, 79 left) (jobs: 0)
                                     
Done: 26% (28/107, 79 left) (jobs: 0)
                                     
Done: 26% (28/107, 79 left) (jobs: 0)
                                     
Done: 15% (28/183, 155 left) (jobs: 0)
                                      
Done: 15% (28/183, 155 left) (jobs: 0)
                                      
Done: 15% (28/183, 155 left) (jobs: 0)
                                      
Done: 15% (28/183, 155 left) (jobs: 0)
                                      
Done: 19% (84/436, 352 left) (jobs: 0)
                                      
Done: 22% (103/461, 358 left) (jobs: 0)
                                       
Done: 21% (103/473, 370 left) (jobs: 0)
                                       
Done: 21% (103/473, 370 left) (jobs: 0)
                                       
Done: 21% (103/473, 370 left) (jobs: 0)
                                       
Done: 20% (103/493, 390 left) (jobs: 0)
                                       
Done: 20% (103/493, 390 left) (jobs: 0)
                                       
Done: 13% (103/746, 643 left) (jobs: 0)
                                       
Done: 13% (103/746, 643 left) (jobs: 0)
                                       
Done: 13% (103/746, 643 left) (jobs: 0)
                                       
Done: 13% (103/746, 643 left) (jobs: 0)
                                       
Done: 10% (103/961, 858 left) (jobs: 0)
                                       
Done: 10% (103/961, 858 left) (jobs: 0)
                                       
Done: 10% (103/961, 858 left) (jobs: 0)
                                       
Done: 17% (168/986, 818 left) (jobs: 0)
                                       
Done: 17% (168/986, 818 left) (jobs: 0)
                                       
Done: 17% (171/998, 827 left) (jobs: 0)
                                       
Done: 10% (176/1702, 1526 left) (jobs: 0)
                                         
Done: 10% (176/1702, 1526 left) (jobs: 0)
                                         
Done: 10% (176/1702, 1526 left) (jobs: 0)
                                         
Done: 13% (234/1702, 1468 left) (jobs: 0)
                                         
Done: 13% (234/1702, 1468 left) (jobs: 0)
                                         
Done: 13% (234/1702, 1468 left) (jobs: 0)
                                         
Done: 13% (234/1702, 1468 left) (jobs: 0)
                                         
Done: 25% (494/1942, 1448 left) (jobs: 0)
                                         
Done: 24% (523/2107, 1584 left) (jobs: 0)
                                         
Done: 25% (545/2112, 1567 left) (jobs: 0)
                                         
Done: 22% (557/2468, 1911 left) (jobs: 0)
                                         
Done: 22% (557/2468, 1911 left) (jobs: 0)
                                         
Done: 31% (770/2468, 1698 left) (jobs: 0)
                                         
Done: 31% (770/2468, 1698 left) (jobs: 0)
                                         
Done: 33% (840/2483, 1643 left) (jobs: 0)
                                         
Done: 33% (849/2510, 1661 left) (jobs: 0)
                                         
Done: 33% (849/2510, 1661 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 99% (2518/2530, 12 left) (jobs: 0)
                                        
Done: 91% (2528/2766, 238 left) (jobs: 0)
                                         
Done: 97% (2736/2794, 58 left) (jobs: 0)
                                        
Done: 99% (2793/2794, 1 left) (jobs: 0)
                                       
Done: 99% (2793/2805, 12 left) (jobs: 0)
                                        
Done: 99% (2798/2805, 7 left) (jobs: 0)
                                       
Done: 63% (2798/4381, 1583 left) (jobs: 0)
                                          
Done: 63% (2798/4381, 1583 left) (jobs: 0)
                                          
Done: 63% (2798/4381, 1583 left) (jobs: 0)
                                          
Done: 63% (2798/4381, 1583 left) (jobs: 0)
                                          
Done: 86% (3813/4386, 573 left) (jobs: 0)
                                         
Done: 86% (3813/4386, 573 left) (jobs: 0)
                                         
Done: 86% (3813/4386, 573 left) (jobs: 0)
                                         
Done: 92% (4061/4386, 325 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 96% (4543/4685, 142 left) (jobs: 0)
                                         
Done: 98% (4637/4685, 48 left) (jobs: 0)
                                        
Done: 87% (4637/5280, 643 left) (jobs: 0)
                                         
Done: 87% (4637/5280, 643 left) (jobs: 0)
                                         
Done: 87% (4637/5280, 643 left) (jobs: 0)
                                         
Done: 98% (5223/5284, 61 left) (jobs: 0)
                                        
Done: 99% (5263/5296, 33 left) (jobs: 0)
                                        
[2023-10-01 14:48:59,153] [INFO] ['make', 'test']
abort: no repository found in '/home/spider/Documents/pyre-check/source' (.hg not found)
Scanned 0 directories
                     
Done: 0% (0/0, 0 left) (jobs: 0)
                                
Done: 0% (0/2, 2 left) (jobs: 0)
                                
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 50% (2/4, 2 left) (jobs: 0)
                                 
Done: 50% (2/4, 2 left) (jobs: 0)
                                 
Done: 33% (2/6, 4 left) (jobs: 0)
                                 
Done: 1% (2/107, 105 left) (jobs: 0)
                                    
Done: 1% (2/107, 105 left) (jobs: 0)
                                    
Done: 26% (28/107, 79 left) (jobs: 0)
                                     
Done: 26% (28/107, 79 left) (jobs: 0)
                                     
Done: 26% (28/107, 79 left) (jobs: 0)
                                     
Done: 15% (28/183, 155 left) (jobs: 0)
                                      
Done: 15% (28/183, 155 left) (jobs: 0)
                                      
Done: 15% (28/183, 155 left) (jobs: 0)
                                      
Done: 19% (84/436, 352 left) (jobs: 0)
                                      
Done: 21% (103/473, 370 left) (jobs: 0)
                                       
Done: 21% (103/473, 370 left) (jobs: 0)
                                       
Done: 21% (103/473, 370 left) (jobs: 0)
                                       
Done: 20% (103/493, 390 left) (jobs: 0)
                                       
Done: 20% (103/493, 390 left) (jobs: 0)
                                       
Done: 13% (103/746, 643 left) (jobs: 0)
                                       
Done: 13% (103/746, 643 left) (jobs: 0)
                                       
Done: 13% (103/746, 643 left) (jobs: 0)
                                       
Done: 10% (103/961, 858 left) (jobs: 0)
                                       
Done: 10% (103/961, 858 left) (jobs: 0)
                                       
Done: 10% (103/961, 858 left) (jobs: 0)
                                       
Done: 17% (168/986, 818 left) (jobs: 0)
                                       
Done: 16% (168/998, 830 left) (jobs: 0)
                                       
Done: 10% (176/1702, 1526 left) (jobs: 0)
                                         
Done: 10% (176/1702, 1526 left) (jobs: 0)
                                         
Done: 10% (176/1702, 1526 left) (jobs: 0)
                                         
Done: 13% (234/1702, 1468 left) (jobs: 0)
                                         
Done: 13% (234/1702, 1468 left) (jobs: 0)
                                         
Done: 13% (234/1702, 1468 left) (jobs: 0)
                                         
Done: 27% (477/1731, 1254 left) (jobs: 0)
                                         
Done: 24% (523/2107, 1584 left) (jobs: 0)
                                         
Done: 25% (545/2112, 1567 left) (jobs: 0)
                                         
Done: 25% (547/2113, 1566 left) (jobs: 0)
                                         
Done: 22% (557/2468, 1911 left) (jobs: 0)
                                         
Done: 31% (770/2468, 1698 left) (jobs: 0)
                                         
Done: 33% (840/2474, 1634 left) (jobs: 0)
                                         
Done: 33% (849/2510, 1661 left) (jobs: 0)
                                         
Done: 33% (849/2510, 1661 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 38% (964/2510, 1546 left) (jobs: 0)
                                         
Done: 99% (2508/2510, 2 left) (jobs: 0)
                                       
Done: 91% (2528/2766, 238 left) (jobs: 0)
                                         
Done: 99% (2793/2794, 1 left) (jobs: 0)
                                       
Done: 99% (2793/2794, 1 left) (jobs: 0)
                                       
Done: 63% (2798/4381, 1583 left) (jobs: 0)
                                          
Done: 63% (2798/4381, 1583 left) (jobs: 0)
                                          
Done: 63% (2798/4381, 1583 left) (jobs: 0)
                                          
Done: 63% (2798/4381, 1583 left) (jobs: 0)
                                          
Done: 86% (3813/4386, 573 left) (jobs: 0)
                                         
Done: 86% (3813/4386, 573 left) (jobs: 0)
                                         
Done: 92% (4061/4386, 325 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 86% (4065/4685, 620 left) (jobs: 0)
                                         
Done: 96% (4543/4685, 142 left) (jobs: 0)
                                         
Done: 98% (4637/4685, 48 left) (jobs: 0)
                                        
Done: 87% (4637/5280, 643 left) (jobs: 0)
                                         
Done: 87% (4637/5280, 643 left) (jobs: 0)
                                         
Done: 98% (5234/5288, 54 left) (jobs: 0)
                                        
Scanned 0 directories
                     
Done: 0% (0/0, 0 left) (jobs: 0)
                                
Done: 0% (0/2, 2 left) (jobs: 0)
                                
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (jobs: 0)
                                  
Done: 100% (2/2, 0 left) (j`

**Comment by ProgrammingMuffin** [comment]
I did some digging, and used script output.txt command to write terminal stdout to a file and ran the same command. I was able to extract some logs.

`[2023-10-01 14:48:50,951] [INFO] ['opam', '--cli=2.1', 'install', '--yes', '--assume-depexts', 'base64.3.5.1', 'cmdliner.1.1.1', 'core.v0.15.1', 'core_unix.v0.15.2', 're2.v0.15.0', 'dune.3.7.1', 'yojson.2.0.2', 'jsonm.1.0.2', 'ppx_deriving_yojson.3.7.0', 'ppx_yojson_conv.v0.15.1', 'ounit2.2.2.7', 'menhir.20220210', 'lwt.5.6.1', 'lwt_ppx.2.1.0', 'ounit2-lwt.2.2.7', 'pyre-ast.0.1.9', 'mtime.1.4.0', 'errpy.0.0.9']
[NOTE] Package errpy is already installed (current version is 0.0.9).
[NOTE] Package mtime is already installed (current version is 1.4.0).
[NOTE] Package pyre-ast is already installed (current version is 0.1.9).
[NOTE] Package ounit2-lwt is already installed (current version is 2.2.7).
[NOTE] Package lwt_ppx is already installed (current version is 2.1.0).
[NOTE] Package lwt is already installed (current version is 5.6.1).
[NOTE] Package menhir is already installed (current version is 20220210).
[NOTE] Package ounit2 is already installed (current version is 2.2.7).
[NOTE] Package ppx_yojson_conv is already installed (current version is v0.15.1).
[NOTE] Package ppx_deriving_yojson is already installed (current version is 3.7.0).
[NOTE] Package jsonm is already installed (current version is 1.0.2).
[NOTE] Package yojson is already installed (current version is 2.0.2).
[NOTE] Package dune is already installed (current version is 3.7.1).
[NOTE] Package re2 is already installed (current version is v0.15.0).
[NOTE] Package core_unix is already installed (current version is v0.15.2).
[NOTE] Package core is already installed (current version is v0.15.1).
[NOTE] Package cmdliner is already installed (current version is 1.1.1).
[NOTE] Package base64 is already installed (current version is 3.5.1).
[2023-10-01 14:48:55,388] [INFO] ['make', 'dev']
abort: no repository found in '/home//Documents/pyre-check/source' (.hg not found)
Scanned 0 directories

Scanned 0 directories

Done: 0% (0/0, 0 left) (jobs: 1)

Done: 0% (0/0, 0 left) (jobs: 0)

Done: 0% (0/2, 2 left) (jobs: 0)

Done: 100% (2/2, 0 left) (jobs: 0)

Done: 100% (2/2, 0 left) (jobs: 0)

Done: 100% (2/2, 0 left) (jobs: 0)

Done: 100% (2/2, 0 left) (j`

**Comment by ProgrammingMuffin** [comment]
After several hours of debugging, a simple solution I found was that if you have a computer with minimal specs like mine (a laptop with AMD Ryzen 7 5700U, and 8 gigs of RAM (3.8 gigs of what is used by my desktop OS)), close every possible window. In my case, I closed chrome and the `setup.sh` script worked. 
I've had a similar issue with Java Corretto-17 compilation as well.

Marking this solved.
